engine/core/src/main/java/org/datacleaner/metamodel/datahub/DataHubDataSet.java

```diff
@@ -158,10 +161,36 @@ private String createEncodedUri(DataHubConnection connection, Table table) {
     }
 
     private String getQueryString(Query query, Table table) {
+        final SelectClause selectClause = query.getSelectClause();
+        if (isPlainSelectOfColumns(selectClause)) {
+            addOrderByClauseIfMissing(query, table);            
+        }
         String queryString = query.toSql();
         return queryString.replace(table.getName() + ".", "");
     }
 
+
+    /**
+     * Checks if the select clause is a plain select of columns and not a function e.g. SELECT COUNT (*)
+     * @param selectClause
+     * @return true 
+     */
+    private boolean isPlainSelectOfColumns(SelectClause selectClause) {
+        final boolean hasMultipleSelectItems = selectClause.getItems().size() != 1;
+        if(hasMultipleSelectItems) {
+            return true;
+        }
+        final boolean itemIsNotAFunction = selectClause.getItem(0).getFunction() == null;
```

Comment:
Would be better to iterate through all items and if any of them has a function return false, else return true.